expr/payload.go

```diff
@@ -31,20 +32,57 @@ const (
 	PayloadBaseTransportHeader PayloadBase = unix.NFT_PAYLOAD_TRANSPORT_HEADER
 )
 
+// Possible PayloadCsumType values.
+const (
+	CsumTypeNone PayloadCsumType = unix.NFT_PAYLOAD_CSUM_NONE
+	CsumTypeInet PayloadCsumType = unix.NFT_PAYLOAD_CSUM_INET
+)
+
 type Payload struct {
-	DestRegister uint32
-	Base         PayloadBase
-	Offset       uint32
-	Len          uint32
+	DestRegister   uint32
+	SourceRegister uint32
+	Base           PayloadBase
+	Offset         uint32
+	Len            uint32
+	CsumType       PayloadCsumType
+	CsumOffset     uint32
+	CsumFlags      uint32
 }
 
 func (e *Payload) marshal() ([]byte, error) {
-	data, err := netlink.MarshalAttributes([]netlink.Attribute{
-		{Type: unix.NFTA_PAYLOAD_DREG, Data: binaryutil.BigEndian.PutUint32(e.DestRegister)},
-		{Type: unix.NFTA_PAYLOAD_BASE, Data: binaryutil.BigEndian.PutUint32(uint32(e.Base))},
-		{Type: unix.NFTA_PAYLOAD_OFFSET, Data: binaryutil.BigEndian.PutUint32(e.Offset)},
-		{Type: unix.NFTA_PAYLOAD_LEN, Data: binaryutil.BigEndian.PutUint32(e.Len)},
-	})
+
+	var attrs []netlink.Attribute
+
+	if e.SourceRegister > 0 {
```

Comment:
Don’t we need to distinguish between an unset SourceRegister field and a 0 SourceRegister here, like in the previous PR?

Reply:
sure